kairon/shared/data/data_models.py

```
@@ -1431,15 +1431,33 @@ class FlowTagChangeRequest(BaseModel):
+    sync_options: Union[PetpoojaSyncOptions]
+
+    @root_validator(pre=True)
+    def validate_sync_options_by_provider(cls, values):
+        provider = values.get("connector_type")
+        sync_options = values.get("sync_options")
 
+        if provider == "petpooja":
+            try:
+                values["sync_options"] = PetpoojaSyncOptions(**sync_options)
+            except Exception as e:
+                raise ValueError(f"Invalid sync_options for petpooja: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)